### PR TITLE
Tweak deployment groups metrics

### PR DESCRIPTION
**Pull request by gimaker**
* Always include the full deployment group in all events.
 * For event that either fail or succeed, indicate the result via a
   numeric "success" field.
 * Include a "machine readable" error code for failed tasks.

### CODE REVIEW

**Comment by gimaker** [comment]
@rohansingh @davidxia

**Comment by rohansingh** [comment]
`helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java:812:12: WhitespaceAround: 'catch' is not followed by whitespace.`

File: helios-services/src/main/java/com/spotify/helios/rollingupdate/DeploymentGroupEventFactory.java [hunk 1]
```diff
@@ -37,51 +35,58 @@
...
     return ev;
   }
 
   public Map<String, Object> rollingUpdateDone(final DeploymentGroup deploymentGroup) {
-    final Map<String, Object> ev = Maps.newHashMap();
-    ev.put("eventType", "rollingUpdateDone");
-    ev.put("timestamp", System.currentTimeMillis());
-    ev.put("deploymentGroupName", deploymentGroup.getName());
-    ev.put("state", DeploymentGroupStatus.State.DONE);
+    final Map<String, Object> ev = createEvent("rollingUpdateFinished", deploymentGroup);
```
Comment (davidxia) [inline helios-services/src/main/java/com/spotify/helios/rollingupdate/DeploymentGroupEventFactory.java]: Can we make the method names and eventType the same?

**Comment by gimaker** [comment]
@matslina

File: helios-services/src/main/java/com/spotify/helios/rollingupdate/DeploymentGroupEventFactory.java (hunk 1 above)
Comment (davidxia) [inline helios-services/src/main/java/com/spotify/helios/rollingupdate/DeploymentGroupEventFactory.java]: Oh nvm/

Comment (gimaker) [inline helios-services/src/main/java/com/spotify/helios/rollingupdate/DeploymentGroupEventFactory.java]: No. The eventType is rollingUpdatedFinished both for rollingUpdateDone and rollingUpdateFailed

File: helios-services/src/main/java/com/spotify/helios/rollingupdate/DeploymentGroupEventFactory.java [hunk 2]
```diff
@@ -37,51 +36,68 @@
     HOSTS_CHANGED
   }
 
-  public Map<String, Object> rollingUpdateTaskSucceeded(final DeploymentGroup deploymentGroup,
-                                                        final RolloutTask task) {
+  private Map<String, Object> createEvent(final String eventType,
```
Comment (davidxia) [inline helios-services/src/main/java/com/spotify/helios/rollingupdate/DeploymentGroupEventFactory.java]: Is it worth it to create classes for these objects?

Comment (gimaker) [inline helios-services/src/main/java/com/spotify/helios/rollingupdate/DeploymentGroupEventFactory.java]: For what objects?

A DeploymentGroupEvent class? Probably. But I'd rather do that in a separate commit.

Or specific classes for specific events?

Comment (davidxia) [inline helios-services/src/main/java/com/spotify/helios/rollingupdate/DeploymentGroupEventFactory.java]: Yea a separate commit sounds good. Something like the below?

DeploymentGroupEvent class:
* eventType
* timestamp
* deploymentGroup
* success
* reason
* failedTask

DeploymentGroupTaskEvent class inherits from DeploymentGroupTask and also includes:
* action
* target
* error
* errorCode
* metadata

**Comment by davidxia** [comment]
:+1:

File: helios-services/src/main/java/com/spotify/helios/rollingupdate/DeploymentGroupEventFactory.java (hunk 2 above)
Comment (gimaker) [inline helios-services/src/main/java/com/spotify/helios/rollingupdate/DeploymentGroupEventFactory.java]: Not sure exactly how it'd look (not sure a class hierachy makes sense, especially since things like failedTask make no sense what-so-ever for rollout-task events), but passing in a metadata `Map<String, Object>` is surely kludgy.
